From fd14f7c1ee27a704117389205e1975e9b704e4a9 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 28 Sep 2022 18:46:39 +0300
Subject: [PATCH] [SW-106171] gaudi3: remove vid and rid in tpm init

Currently vid and rid are read in tpm init, but this data is un-used.
Currently this routine would only confirm us that the spi communication
itself is successful, which would anyhow be confirmed by earlier apis
that set the locality.

Hence these apis are redundant and can be removed from rom sequence.
This would save us some time of tpm validation.

Change-Id: I1b2f6d7ad0a9b84190f1f729c28ae55ffe865994
---

diff --git a/arch/gaudi3/arc/tpm2.c b/arch/gaudi3/arc/tpm2.c
index 10c0f38..e730f4d 100644
--- a/arch/gaudi3/arc/tpm2.c
+++ b/arch/gaudi3/arc/tpm2.c
@@ -539,8 +539,6 @@
 int tpm2_init(void)
 {
 	int ret;
-	uint32_t vend_dev;
-	uint8_t rid;
 
 	locality = 0;
 
@@ -554,14 +552,6 @@
 		return ret;
 
 	ret = tpm_tis_spi_request_locality(locality);
-	if (ret)
-		return ret;
-
-	ret = tpm_tis_spi_read32(TPM_DID_VID(locality), &vend_dev);
-	if (ret)
-		return ret;
-
-	ret = tpm_tis_spi_read(TPM_RID(locality), &rid, 1);
 	if (ret)
 		return ret;
 
